pkg/pdutil/region_label.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pdutil
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/pingcap/tiflow/pkg/config"
+	"github.com/pingcap/tiflow/pkg/httputil"
+	"github.com/pingcap/tiflow/pkg/retry"
+	pd "github.com/tikv/pd/client"
+	"go.uber.org/zap"
+)
+
+const (
+	// DefaultMaxRetry is the default max number of verifing region labels.
+	DefaultMaxRetry   = 3
+	regionLabelPrefix = "/pd/api/v1/config/region-label/rules"
+
+	// Split the default rule by `6e000000000000000000f8` to keep metadata region
+	// isolated from the normal data area.
+	addMetaJSON = `{
+		"sets": [
+			{
+				"id": "ticdc/meta",
+				"labels": [
+					{
+						"key": "data-type",
+						"value": "meta"
+					}
+				],
+				"rule_type": "key-range",
+				"data": [
+					{
+						"start_key": "6d00000000000000f8",
+						"end_key": "6e00000000000000f8"
+					}
+				]
+			}
+		]
+	}`
+)
+
+// PDApiClient is api client of Placement Driver.
+type PDApiClient struct {
+	pdClient   pd.Client
+	dialClient *httputil.Client
+}
+
+// NewPDApiClient create a new PlacementClient.
+func NewPDApiClient(ctx context.Context, pdClient pd.Client) (*PDApiClient, error) {
+	conf := config.GetGlobalServerConfig()
+	dialClient, err := httputil.NewClient(conf.Security)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return &PDApiClient{
+		pdClient:   pdClient,
+		dialClient: dialClient,
+	}, nil
+}
+
+// Verify check and update region labels.
+func Verify(ctx context.Context, pdClient pd.Client, maxRetry uint64) error {
```

Comment:
Better to use a more appropriate function name, since this function could **update** the region label, `verify` only describes the `check` behavior, but doesn't contain the `update` behavior.

Besides please add comment about what happens when calling this function multiple times, can also add a unit test to cover the reentrant calling scenario.

Reply:
It is up to the server to decide whether this function is reentrant or not.  So we manually verify the reentrancy of the function and just add a comment to explain this.